django_saml2_auth/views.py

```diff
@@ -249,5 +275,33 @@ def signin(r):
 
 
 def signout(r):
+    saml_client = _get_saml_client(get_current_domain(r))
+    subject_id = _get_subject_id(r.session)
+    idp_entity_id = None
+    try:
+        idp_entity_id = settings.SAML2_AUTH['SAML_CLIENT_SETTINGS']['service']['sp']['idp']
+    except KeyError:
+        return HttpResponseServerError('Idp not defined')
+
+    result = saml_client.do_logout(subject_id, [idp_entity_id], "", None)
+    if len(result) > 1:
+        return HttpResponseServerError('Logout from several sources not supported')
+    for entityid, logout_info in result.items():
+        if isinstance(logout_info, tuple):
+            binding, http_info = logout_info
+            if binding == BINDING_HTTP_POST:
+                body = ''.join(http_info['data'])
+                return HttpResponse(body)
+            elif binding == BINDING_HTTP_REDIRECT:
+                return HttpResponseRedirect(_get_location(http_info))
+            else:
+                return HttpResponseServerError('Failed to log out')
+        else:
+            return finish_signout(r, logout_info)
+
+    return HttpResponseServerError('Logout Binding not supported')
```

Comment:
Add comments here to the if-else cases or to the function docstring to help the reader of the code to understand it. Currently the code raises the following questions, which IMHO should be answered some how:
 * What does it mean if logout_info is tuple or not? (In what cases it is and when not?)
 * What logout_info is, if it's not a tuple? (type and content)
 * What is http_info contains?
 * What are the cases that are being handled when binding is POST or REDIRECT and why they are handled the way they are?
 * Why other bindings are raising a failure?
 * Why `finish_signout` is not always called?
 * Why user is not logged out locally in most of the cases?